go/vt/vttablet/tabletmanager/vreplication/table_plan_builder.go

```diff
@@ -669,9 +675,18 @@ func (tpb *tablePlanBuilder) generateValuesPart(buf *sqlparser.TrackedBuffer, bv
 		separator = ","
 		switch cexpr.operation {
 		case opExpr:
-			if cexpr.colType == querypb.Type_JSON {
+			switch cexpr.colType {
+			case querypb.Type_JSON:
 				buf.Myprintf("convert(%v using utf8mb4)", cexpr.expr)
-			} else {
+			case querypb.Type_DATETIME:
+				sourceTZ := tpb.source.SourceTimeZone
+				targetTZ := tpb.source.TargetTimeZone
+				if sourceTZ != "" && targetTZ != "" {
+					buf.Myprintf("convert_tz(%v, '%s', '%s')", cexpr.expr, sourceTZ, targetTZ)
```

Comment:
I don't see where we ever validate that the user specified a valid named time zone for the flag? And if the specified something invalid:
```
mysql> select convert_tz("2022-04-26 00:02:33", "US/Eastern", "UTC");
+--------------------------------------------------------+
| convert_tz("2022-04-26 00:02:33", "US/Eastern", "UTC") |
+--------------------------------------------------------+
| 2022-04-26 04:02:33                                    |
+--------------------------------------------------------+
1 row in set (0.00 sec)

mysql> select convert_tz("2022-04-26 00:02:33", "US/Nosir", "UTC");
+------------------------------------------------------+
| convert_tz("2022-04-26 00:02:33", "US/Nosir", "UTC") |
+------------------------------------------------------+
| NULL                                                 |
+------------------------------------------------------+
1 row in set (0.00 sec)
```

I'm assuming that VDiff will catch this? Still, IMO we should validate the flag before we get here. We can be sure that the target MySQL *both* has the time zone info tables loaded, *and* that the user specified a valid one this way: 
```
mysql> SET session time_zone = 'US/Nosir';
ERROR 1298 (HY000): Unknown or incorrect time zone: 'US/Nosir'

mysql> SET session time_zone = 'US/Pacific';
Query OK, 0 rows affected (0.00 sec)
```

Reply:
Added a validation, so that MoveTables will return an error to the user if time zone is not found on the _current_ target primaries. Note that it is always possible for a new tablet to become the primary: we can, of course, check every current replica, but user can always add a new replica without loading the time zone ...